ca_tools/scripts/pid_controller.py

```diff
@@ -0,0 +1,238 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2013 PAL Robotics SL.
+# Released under the BSD License.
+#
+# Authors:
+#   * Ramiro Serra
+
+import math
+import signal
+import sys
+from math import pi
+import rospy
+from geometry_msgs.msg import Twist
+from geometry_msgs.msg import Pose2D
+from geometry_msgs.msg import PoseStamped
+from geometry_msgs.msg import Point
+from nav_msgs.msg import Path
+from nav_msgs.msg import Odometry
+from std_msgs.msg import Float64
+from tf.transformations import euler_from_quaternion
+
+ANGLE_TOLERANCE = 0.1
+ANGULAR_VEL = 1.0
+PID_ANGULAR_EFFORT_ADJUST = 5.0  # factor that divides the effort to send the cmd_vel command
+TURNING = 2
+FORWARD = 3
+STOP_FORWARD = 4
+DISTANCE_TOLERANCE = 0.2
+PID_LINEAL_EFFORT_ADJUST = 10.0
+
+pubtopic = "/create1/cmd_vel"
+subtopic = "/create1/gts"
+
+
+def signal_handler(signal, frame, queue_size=1):
+  # This will execute when Ctrl-c is pressed
+    pubtopic = "/create1/cmd_vel"
+    pub = rospy.Publisher(pubtopic, Twist, queue_size=1)
+    aux = Twist()
+    pub.publish(aux)
+    sys.exit(0)
+
+
+class Ctrl_Node:
+    """A class that inits a node, in order to make the robot move to 4 fixed points, drawing a square
+    """
+
+    def __init__(self):
+
+        rospy.init_node('control_node')
+
+        #suscribers and publishers
+        self._my_pid_angular_state_pub = rospy.Publisher(
+            "/angular/state", Float64, queue_size=1)
+        self._my_pub = rospy.Publisher(pubtopic, Twist, queue_size=10)
+        self._my_pid_lineal_state_pub = rospy.Publisher(
+            "/lineal/state", Float64, queue_size=1)
+        self._my_pid_angular_effort_sub = rospy.Subscriber(
+            "/lineal/control_effort", Float64, self._lineal_effort_callback)
+        self._my_pid_lineal_setpoint_pub=rospy.Publisher("/lineal/setpoint",Float64,queue_size=10)
+        self._my_sub = rospy.Subscriber(subtopic, Odometry, self._callback)
+        self._my_pid_angular_effort_sub = rospy.Subscriber(
+            "/angular/control_effort", Float64, self._angular_effort_callback)
+        self._my_path_publisher = rospy.Publisher(
+            "/create1/my_path", Path, queue_size=10)
+        self._my_path = Path()
+        self._my_pid_angular_setpoint_pub = rospy.Publisher("/angular/setpoint", Float64, queue_size=1)
+
+        self._my_goals = []
+        self._path_threshold = 60 #Throttle the path pub
+        #CREATE GOALS
+        for i in range(4):
+            self._my_goals.append(Pose2D())
+        self._my_goals[1].x = 2.0
+        self._my_goals[2].x = 2.0
+        self._my_goals[2].y = 2.0
+        self._my_goals[3].y = 2.0
+
+        self._lineal_state=Float64()
```

Comment:
Why do you need this? Either initialize with a value of `None` if you want it to be explicitly uninitialized